lib/class-wp-theme-json-gutenberg.php

```diff
@@ -669,6 +684,9 @@ private static function compute_preset_vars( $settings ) {
 			$preset_per_origin = _wp_array_get( $settings, $preset['path'], array() );
 			$preset_by_slug    = self::get_merged_preset_by_slug( $preset_per_origin, $preset['value_key'] );
 			foreach ( $preset_by_slug as $slug => $value ) {
+				if ( ! empty( $preset['value_wrapper'] ) ) {
+					$value = str_replace( '$s', $value, $preset['value_wrapper'] );
+				}
```

Comment:
Here's an example of what I'm saying in https://github.com/WordPress/gutenberg/pull/34073/files/ec08ac957102cfa825e455224fd31ea070a12fdb#r688683322.

```php
/* lib/class-wp-theme-json-gutenberg.php */
		array(
			'path'          => array( 'color', 'duotone' ),
			'value_key'     => array( 'slug' ),
			'css_var_infix' => 'duotone',
			'value_wrapper' => 'url(#wp-duotone-filter-%s)',
			'classes'       => array(
				array(
					'class_suffix'  => 'duotone',
					'property_name' => 'filter',
				),
			),
		),
```

```suggestion
				if ( ! empty( $preset['value_format'] ) ) {
					$value = vsprintf( $preset['value_format'], is_array( $value ) ? $value : array( $value ) );
				}
```

Reply:
Thanks. I'm not sure what changes need to be made to `get_merged_preset_by_slug`. What does that function do?

Reply:
It converts from `value_key` to the actual value, so it'll have to be updated to support arrays.

```diff
-               $result[ preg_replace( '/\s+/', '-', $preset['slug'] ) ] = $preset[ $value_key ];
+               $result[ preg_replace( '/\s+/', '-', $preset['slug'] ) ] = is_array( $value_key )
+                   ? array_map( function ( $key ) use ( $preset ) { return $preset[ $key ]; }, $value_key )
+                   : $preset[ $value_key ];
```